File: app/src/main/java/com/sortedqueue/programmercreek/fragments/ProgramWikiPagerAdapter.java
package com.sortedqueue.programmercreek.fragments;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.sortedqueue.programmercreek.database.ProgramWiki;
import com.sortedqueue.programmercreek.database.WikiModel;

import java.util.ArrayList;

/**
 * Created by Alok Omkar on 2016-12-31.
 */

public class ProgramWikiPagerAdapter extends FragmentPagerAdapter {

    private ArrayList<ProgramWikiFragment> ProgramWikiFragments = new ArrayList<>();
    private ArrayList<WikiModel> programWikis;

    public ProgramWikiPagerAdapter(FragmentManager fm, ArrayList<WikiModel> programWikis) {
        super(fm);
        this.programWikis = programWikis;
        for( WikiModel programWiki : programWikis ) {
            ProgramWikiFragment ProgramWikiFragment = new ProgramWikiFragment();
            ProgramWikiFragment.setParams( programWiki );
            ProgramWikiFragments.add(ProgramWikiFragment);
        }


    }

    @Override
    public Fragment getItem(int position) {
        return ProgramWikiFragments.get(position);
    }

    @Override
    public int getCount() {
        return programWikis.size();
    }
}

File: app/src/main/java/com/sortedqueue/programmercreek/adapter/ProgramWikiPagerAdapter.kt
package com.sortedqueue.programmercreek.adapter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

import com.sortedqueue.programmercreek.database.WikiModel
import com.sortedqueue.programmercreek.fragments.ProgramWikiFragment

import java.util.ArrayList

/**
 * Created by Alok Omkar on 2016-12-31.
 */

class ProgramWikiPagerAdapter(fm: FragmentManager, private val programWikis: ArrayList<WikiModel>) : FragmentPagerAdapter(fm) {

    private val ProgramWikiFragments = ArrayList<ProgramWikiFragment>()

    init {
        for (programWiki in programWikis) {
            val ProgramWikiFragment = ProgramWikiFragment()
            ProgramWikiFragment.setParams(programWiki)
            ProgramWikiFragments.add(ProgramWikiFragment)
        }


    }

    override fun getItem(position: Int): Fragment {
        return ProgramWikiFragments[position]
    }

    override fun getCount(): Int {
        return programWikis.size
    }
}
